mmf/utils/checkpoint.py

```diff
@@ -437,23 +453,29 @@ def save(self, update, iteration=None, update_best=False):
             git_metadata_dict = self._get_vcs_fields()
             ckpt.update(git_metadata_dict)
 
+        logger.info("Saving  checkpoint")
         with PathManager.open(ckpt_filepath, "wb") as f:
-            torch.save(ckpt, f)
+            self.save_func()(ckpt, f)
```

Comment:
this looks ugly. I'd rewrite it:

```
@property
def save_func(self):
  ...
```

then you can call it like this:

`self.save_func(chpt, g)`